ArduPlane/compassmot.cpp

```diff
@@ -0,0 +1,248 @@
+    in_calibration = true;
+
+    float current;
+
+    // default compensation type to use current if possible
+    if (battery.current_amps(current)) {
+        comp_type = AP_COMPASS_MOT_COMP_CURRENT;
+    } else {
+        comp_type = AP_COMPASS_MOT_COMP_THROTTLE;
+    }
+
+    // send back initial ACK
+    mavlink_msg_command_ack_send(gcs_chan.get_chan(), MAV_CMD_PREFLIGHT_CALIBRATION,0);
+
+    // flash leds
+    AP_Notify::flags.esc_calibration = true;
+
+    // warn user we are starting calibration
+    gcs_chan.send_text(MAV_SEVERITY_INFO, "Starting Calibration");
+
+    // inform what type of compensation we are attempting
+    if (comp_type == AP_COMPASS_MOT_COMP_CURRENT) {
+        gcs_chan.send_text(MAV_SEVERITY_INFO, "Current");
+    } else {
+        gcs_chan.send_text(MAV_SEVERITY_INFO, "Throttle");
+    }
+
+    // disable throttle failsafe
+    g.throttle_fs_enabled = 0;
+  
+    // disable motor compensation
+    compass.motor_compensation_type(AP_COMPASS_MOT_COMP_DISABLED);
+    for (uint8_t i=0; i<compass.get_count(); i++) {
+        compass.set_motor_compensation(i, Vector3f(0,0,0));
+    }
+
+    // get initial compass readings
+    compass.read();
+
+    // store initial x,y,z compass values
+    // initialise interference percentage
+    for (uint8_t i=0; i<compass.get_count(); i++) {
+        compass_base[i] = compass.get_field(i);
+        interference_pct[i] = 0.0f;
+    }
+
+    EXPECT_DELAY_MS(5000);
+
+    // enable motors and pass through throttle
+    init_rc_out_main();
+    hal.util->set_soft_armed(true);
+
+    // initialise run time
+    last_run_time = millis();
+    last_send_time = millis();
+
+    // main run while there is no user input and the compass is healthy
+    while (command_ack_start == command_ack_counter && compass.healthy()) {
+        EXPECT_DELAY_MS(5000);
+
+        // 50hz loop
+        if (millis() - last_run_time < 20) {
+            hal.scheduler->delay(5);
+            continue;
+        }
+        last_run_time = millis();
+
+        // read the radio input
+        read_radio();
+
+        // pass through throttle to motors
+        SRV_Channels::cork();
+        SRV_Channels::set_output_scaled(SRV_Channel::k_throttle, get_throttle_input(true));
+        SRV_Channels::push();
+
+        // read some compass values
+        compass.read();
+
+        // read current
+        battery.read();
+
+        // calculate scaling for throttle
+        throttle_pct = (float)channel_throttle->get_control_in() / 1000.0f;
+        throttle_pct = constrain_float(throttle_pct,0.0f,1.0f);
+
+        if (!battery.current_amps(current)) {
+        current = 0;
+        }
+        current_amps_max = MAX(current_amps_max, current);
+
+        // if throttle is near zero, update base x,y,z values
+        if (throttle_pct <= 0.0f) {
+            for (uint8_t i=0; i<compass.get_count(); i++) {
+            compass_base[i] = compass_base[i] * 0.99f + compass.get_field(i) * 0.01f;
```

Comment:
```suggestion
                compass_base[i] = compass_base[i] * 0.99f + compass.get_field(i) * 0.01f;
            }
```
There are several indentation issues, like this one